x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,238 @@
+	poolAddr, err := sdk.AccAddressFromBech32(pool.PoolAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+	poolBalance := k.bankKeeper.GetBalance(ctx, poolAddr, assets.MicroMedDenom)
+	if pool.IsPaidDeposit {
+		return &poolBalance.Amount, nil
+	} else {
+		amount := poolBalance.Amount.Sub(pool.Deposit.Amount)
+		return &amount, nil
+	}
+}
+
+func (k Keeper) executeRevenueDistribute(ctx sdk.Context, pool *types.Pool) error {
+	availablePoolCoinAmount, err := k.getAvailablePoolCoinAmount(ctx, pool)
+	if err != nil {
+		return err
+	}
+
+	// calculate the revenue to be sent to each seller
+	eachDistributionAmount := k.getEachDistributionAmount(pool)
+
+	poolAddress, err := sdk.AccAddressFromBech32(pool.GetPoolAddress())
+	if err != nil {
+		return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+
+	salesHistories := k.ListSalesHistory(ctx, pool.PoolId, pool.Round)
+	for _, history := range salesHistories {
+		// if there is no coin available, proceed no further.
+		if availablePoolCoinAmount.Equal(sdk.NewInt(0)) {
+			break
+		}
+
+		paidCoinAmount := history.PaidCoin.Amount
+		dataCount := sdk.NewInt(int64(history.DataCount()))
+		distributedAmount := eachDistributionAmount.Mul(dataCount)
+
+		// if it has already been distributed, pass it.
+		if distributedAmount.Equal(paidCoinAmount) {
+			continue
+		}
+
+		paymentAmount := distributedAmount.Sub(paidCoinAmount)
+		// If the transferable amount is less than the payable amount, it is replaced with the transferable amount.
+		if availablePoolCoinAmount.LT(paymentAmount) {
+			paymentAmount = *availablePoolCoinAmount
+		}
+		sellerAddr, err := sdk.AccAddressFromBech32(history.SellerAddress)
+		if err != nil {
+			return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+		}
+		paymentCoin := sdk.NewCoin(assets.MicroMedDenom, paymentAmount)
+		err = k.bankKeeper.SendCoins(
+			ctx,
+			poolAddress,
+			sellerAddr,
+			sdk.NewCoins(paymentCoin))
+		if err != nil {
+			return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+		}
+		*history.PaidCoin = history.PaidCoin.Add(paymentCoin)
+
+		// Subtract the amount paid from the transferable amount.
+		*availablePoolCoinAmount = availablePoolCoinAmount.Sub(paymentAmount)
+
+		k.SetSalesHistory(ctx, history)
+	}
+	return nil
+}
+
+func (k Keeper) getEachDistributionAmount(pool *types.Pool) sdk.Int {
+	maxNftSupply := sdk.NewIntFromUint64(pool.PoolParams.MaxNftSupply).ToDec()
+	nftPriceAmount := pool.PoolParams.NftPrice.Amount.ToDec()
+	totalNftPriceAmount := maxNftSupply.Mul(nftPriceAmount)
+
+	numIssuedNFTs := sdk.NewIntFromUint64(pool.NumIssuedNfts).ToDec()
+	targetNumData := sdk.NewIntFromUint64(pool.PoolParams.TargetNumData).ToDec()
+
+	distributeRate := numIssuedNFTs.Quo(maxNftSupply)
+	totalNftPriceAmount = totalNftPriceAmount.Mul(distributeRate)
+
+	return totalNftPriceAmount.Quo(targetNumData).TruncateInt()
+}
```

Comment:
This `getEachDistributionAmount()` can be simplified.
```
EachDistributionAmount = (NFTPrice * numIssuedNFTs) / targetNumData
```
which is the amount to be distributed per data.

Reply:
Oh! There was a simple way, but I implemented it in a complicated way.
Thank you. 😊
I fixed it. (https://github.com/medibloc/panacea-core/commit/da195915ca26f7e51121dc3cc674a287c06dcbdb)